disco/disco.go

```diff
@@ -49,6 +49,16 @@ type Disco struct {
 	Transport http.RoundTripper
 }
 
+// ErrServiceDiscoveryNetworkRequest represents the error that occurs when
+// the service discovery fails for an unknown network problem.
+type ErrServiceDiscoveryNetworkRequest struct {
+	err error
+}
+
+func (e *ErrServiceDiscoveryNetworkRequest) Error() string {
```

Comment:
💅 no need for a pointer receiver on this method

```suggestion
func (e ErrServiceDiscoveryNetworkRequest) Error() string {
```